plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/notation/crowsfoot/CrowsFootDiagramNotation.java

```diff
@@ -56,34 +63,39 @@ public void applyNotationForArrows(PolylineConnection conn, ERDAssociation assoc
             association.getTargetEntity() instanceof ERDEntity) {
             // source - 1..n
             try {
-                ERDEntity src = (ERDEntity) association.getSourceEntity();
-                DBSEntity entity = src.getObject();
-                VoidProgressMonitor monitor = new VoidProgressMonitor();
-                Collection<? extends DBSTableIndex> indexes = ((DBSTable) entity).getIndexes(monitor);
-                if (!CommonUtils.isEmpty(indexes)) {
-                    // get index for require source attributes
-                    List<ERDEntityAttribute> erdSourceAttributes = association.getSourceAttributes();
-                    List<DBSEntityAttribute> attributes = erdSourceAttributes.stream()
-                        .map(ERDEntityAttribute::getObject)
-                        .toList();
-                    if (DBUtils.isUniqueIndexForAttributes(monitor, attributes, entity)) {
-                        createSourceDecorator(conn, bckColor, frgColor, ERDAssociationType.ONE_ONLY, LABEL_1);
-                    } else {
-                        createSourceDecorator(conn, bckColor, frgColor, ERDAssociationType.ONE_OR_MANY, LABEL_1_TO_N);
+                UIUtils.runInProgressService(monitor -> {
```

Comment:
You cannot create progress service for each part. It is super slow.
It can be created only once per entire diagram layout.